server/manager/src/main/java/org/apache/accumulo/manager/split/Splitter.java

```diff
@@ -84,52 +74,34 @@ public int hashCode() {
 
   LoadingCache<CacheKey,FileInfo> splitFileCache;
 
-  public static int weigh(KeyExtent keyExtent) {
-    int size = 0;
-    size += keyExtent.tableId().toString().length();
-    if (keyExtent.endRow() != null) {
-      size += keyExtent.endRow().getLength();
-    }
-    if (keyExtent.prevEndRow() != null) {
-      size += keyExtent.prevEndRow().getLength();
-    }
-    return size;
-  }
-
   public Splitter(ServerContext context) {
-    this.splitExecutor = context.threadPools().createExecutorService(context.getConfiguration(),
-        Property.MANAGER_SPLIT_WORKER_THREADS, true);
+    int numThreads = context.getConfiguration().getCount(Property.MANAGER_SPLIT_WORKER_THREADS);
+    // Set up thread pool that constrains the amount of task it queues and when full discards task.
+    // The purpose of this is to avoid reading lots of data into memory if lots of tablets need to
+    // split.
+    BlockingQueue<Runnable> queue = new ArrayBlockingQueue<>(10000);
+    this.splitExecutor = context.threadPools().createThreadPool(numThreads, numThreads, 0,
```

Comment:
You could remove the branch in ThreadPools.createExecutorService that handles the property MANAGER_SPLIT_WORKER_THREADS.

Reply:
removed 0f1f2fa